2758 fix compilation tests

hiker opened the pull request:
A few small changes to fix #2758.

hiker commented:
Ready for review. The link check failure (to https://fortranwiki.org/fortran/show/Intrinsic+procedures) is not caused by this PR, the site seems unavailable atm.

arporter reviewed (changes requested):
Thanks very much for the quick fix Joerg. All looks good and nice to see how encapsulated the problem was.
However, I've realised that we have `lib/extract/{netcdf,standalone}/nemo` and the READMEs in those directories talk about the NEMO API. Would you mind renaming those directories (perhaps s/nemo/no_psykal/ or something) and updating the READMEs too?

hiker commented:
I completely missed that PR. By now, Sergi has renamed the directory. I've fixed the references to NEMO in the files that were still there. The only exception are references to `examples`, where we still have a NEMO directory.

hiker commented on .github/workflows/compilation.yml:
```diff
@@ -51,13 +51,13 @@ on:
   push
```
I have no idea how it shows up for you. Looking at "Files Changed" above, it doesn't show up. Looing at `git blame`, I am also not seeing anything from me:
```
psyclone/.github/workflows$ git blame compilation.yml | grep -v Andrew | grep -v Nobre | grep -v Siso
psyclone/.github/workflows$ 
```
Only Nuno, Sergi, and you show up in blame :( Where do you see a change?

hiker commented:
@arporter: I've updated it to latest master. Could you tell me where you see a modification to a yaml file?

arporter commented:
I've triggered the integration tests.

arporter commented:
The compilation tests failed (https://github.com/stfc/PSyclone-mirror/actions/runs/12369603001/job/34521856319) :-( Would you mind taking a look @hiker?

hiker commented:
It worked for me, because I had a jinja-procssed file in `extract/tandalone/lfric`, which is missing in a clean checkout.

What I don't understand: how on earth did that pass even the normal CI? For me (once I cleaned that directory), normal (non-compilation) tests failed, because the file was missing :(  No idea. I've removed the preprocessed file  `extract/standalone/read_kernel_data_mod.f90`, and added it to `extract/standalone/lfric`. Will test CI and then compilation test (which again worked for me, and I tried to clean my directories properly :) )

Note that gocean extraction is not affected - at this point gocean does not inline the dependent modules (and therefore does not need this file at all).
